doc/designs/quic-design/tx-packetiser.md

```diff
@@ -0,0 +1,435 @@
+TX Packetiser
+=============
+
+This module creates frames from the application data obtained from
+the application.  It also receives CRYPTO frames from the TLS Handshake
+Record Layer and ACK frames from the ACK Handling And Loss Detector
+subsystem.
+
+The packetiser also deals with the flow and congestion controllers.
+
+Creation & Destruction
+----------------------
+
+```c
+struct ossl_quic_tx_packetiser_st {
+    QUIC_CONNECTION *conn;
+};
+
+_owur typedef struct ossl_quic_tx_packetiser_st OSSL_QUIC_TX_PACKETISER;
+
+OSSL_QUIC_TX_PACKETISER ossl_quic_tx_packetiser_new(QUIC_CONNECTION *conn);
+void ossl_quic_tx_packetiser_free(OSSL_QUIC_TX_PACKETISER *tx);
+```
+
+Structures
+----------
+
+### Connection
+
+Represented by an QUIC_CONNECTION object.
+
+### Stream
+
+Represented by an QUIC_STREAM object.
+
+As per [RFC 9000 2.3 Stream Prioritization], streams should contain a priority
+provided by the calling application.  For MVP, this is not required to be
+implemented because only one stream is supported.  However, packets being
+retransmitted should be preferentially sent as noted in
+[RFC 9000 13.3 Retransmission of Information].
+
+```c
+void OSSL_QUIC_STREAM_set_priority(SSL *stream, uint32_t priority);
+uint32_t OSSL_QUIC_STREAM_get_priority(SSL *stream);
```

Comment:
> Aside: Are we using OSSL_QUIC_STREAM or QUIC_STREAM for the stream object. I've seen both used.

In what context?  For structures and types, QUIC_ is okay.  For functions, they should begin ossl_ IMO.

Reply:
So, the object would be `QUIC_STREAM`, but any function that uses it would be `OSSL_QUIC_STREAM_whatever(QUIC_STREAM *stream, ...)` (or should that be `ossl_quic_stream_whatever(QUIC_STREAM *stream, ...)` for internal things)?

Reply:
I think so.